### PR TITLE
Options: sync mailserver settings

### PR DESCRIPTION
**Pull request by simison**
- Sync changes in these options:
  - `mailserver_url`
  - `mailserver_login` — Not syncing contents, only the option name
  - `mailserver_pass`  — Not syncing contents, only the option name
  - `mailserver_port`

Adds new list for contentless options and a new `jetpack_sync_options_contentless` filter for them.

Options listed in the contentless list will be synced but without content. Only the option name is sent.

Instead of:

```
(
    [0] => mailserver_login
    [1] => foo
    [2] => bar
)
```

Syncs this:
```
(
    [0] => mailserver_login
    [1] =>
    [2] =>
)
```

Synced information will be shown in the Activity Log.

`default_email_category` is already synced in `master`.

## Testing

- Modify "Post via email" settings via `/wp-admin/options-writing.php`

    image
- Observe values get synced
- Observe `mailserver_login` and `mailserver_pass` have empty strings as values.
- Tests should pass: `phpunit --filter=WP_Test_Jetpack_Sync_Options`

### CODE REVIEW

File: tests/php/sync/test_class.jetpack-sync-options.php [hunk 1]
```diff
@@ -179,14 +179,18 @@ public function test_sync_default_options() {
 			'uploads_use_yearmonth_folders'        => '0',
 			'date_format'                          => '0',
 			'time_format'                          => '0',
+			'mailserver_url'                       => 'pineapple.example.com',
+			'mailserver_login'                     => '',
+			'mailserver_pass'                      => '',
```
Comment (simison) [inline tests/php/sync/test_class.jetpack-sync-options.php]: If I would add values like "pineapple" for `mailserver_login` and `mailserver_pass `, I would need to fetch the list of contentless options and create an exception for any options on that list when their values are tested in:

```php
foreach ( $options as $option_name => $value ) {
  $this->assertOptionIsSynced( $option_name, $value );
}

File: sync/class.jetpack-sync-defaults.php [hunk 2]
```diff
@@ -137,6 +141,25 @@ public static function get_options_whitelist() {
...
+		/**
+		 * Filter the list of WordPress options that should be synced without content
+		 *
+		 * @module sync
+		 *
+		 * @since 6.1
+		 *
+		 * @param array The list of options synced without content.
+		 */
+		return apply_filters( 'jetpack_sync_options_contentless', self::$default_options_contentless );
```
Comment (lezama) [inline sync/class.jetpack-sync-defaults.php]: like the filter 👍

**Comment by oskosk** [comment]
@simison this one needs a rebase. Please relable as needs review afterwards.

**Comment by simison** [comment]
Rebased :thumbsup:

**Pull request merged**